metrics/zio-metrics/src/main/scala/sttp/tapir/server/metrics/zio/ZioMetrics.scala

```diff
@@ -0,0 +1,178 @@
+package sttp.tapir.server.metrics.zio
+
+import sttp.tapir._
+import sttp.tapir.model.ServerRequest
+import sttp.tapir.server.interceptor.metrics.MetricsRequestInterceptor
+import sttp.tapir.server.metrics.{EndpointMetric, Metric, MetricLabels}
+import sttp.tapir.server.model.ServerResponse
+import zio._
+import zio.metrics.Metric._
+import zio.metrics.MetricKeyType.Histogram.Boundaries
+import zio.metrics.MetricLabel
+
+import java.time.{Clock, Duration, Instant}
+
+case class ZioMetrics[F[_]](namespace: String, metrics: List[Metric[F, _]] = List.empty) {
+  import ZioMetrics._
+
+  /** Registers a `$namespace.request_active.count|c|#path, method` counter (assuming default labels). */
+  def addRequestsActive(labels: MetricLabels = MetricLabels.Default): ZioMetrics[F] =
+    copy(metrics = metrics :+ requestActive(namespace, labels))
+
+  /** Registers a `request_total{path, method, status}` counter (assuming default labels). */
+  def addRequestsTotal(labels: MetricLabels = MetricLabels.Default): ZioMetrics[F] =
+    copy(metrics = metrics :+ requestTotal(namespace, labels))
+
+  /** Registers a `request_duration_seconds{path, method, status, phase}` histogram (assuming default labels). */
+  def addRequestsDuration(labels: MetricLabels = MetricLabels.Default): ZioMetrics[F] =
+    copy(metrics = metrics :+ requestDuration(namespace, labels))
+
+  /** Registers a custom metric. */
+  def addCustom(m: Metric[F, _]): ZioMetrics[F] = copy(metrics = metrics :+ m)
+
+
+  def metricsInterceptor(ignoreEndpoints: Seq[AnyEndpoint] = Seq.empty): MetricsRequestInterceptor[F] =
+    new MetricsRequestInterceptor[F](metrics, ignoreEndpoints)
+}
+
+object ZioMetrics {
+  val DEFAULT_NAMESPACE: String = "tapir"
```

Comment:
[minor] by convention constants in scala are camel case, e.g. `DefaultNamespace`